### DISCUSSION TITLE
Can't load environment variables from CI/CD pipeline

### DISCUSSION THREAD

**Discussion by OfficePop** [gatsbyjs/gatsby#31933] [Help] [answered]
I am having trouble getting my gatsby-config file to load an environment variable defined in my CI/CD pipeline file (Using Jenkins, for reference).

I define the following environment variable in my jenkinsfile:

```
environment {
     CONTENTFUL_ENV = "${env.GIT_BRANCH}"
}
```

In gatsby-config I try to call this variable like so:

```
module.exports = {
  siteMetadata: {
    title: `<Website Title>`,
  },
  plugins: [
    {
      resolve: `gatsby-source-contentful`,
      options: {
        spaceId: `******`,
        accessToken: `*****`,
        environment: process.env.CONTENTFUL_ENV,
      },
    },
  ],
};
```

The idea is for the environment to match whatever the pipeline is running in (dev or master), but this build keeps failing because it doesn't appear to be reading the variable define in the pipeline document. Is there something i'm missing?

**Comment by KyleAMathews** [comment, marked as answer]
This looks like a Jenkins issue. To help debug you could add a console.log(process.env) to the file.
